File: app/src/main/java/com/cookandroid/torchapp/FlashTread.java
package com.cookandroid.torchapp;

import android.content.Context;
import android.os.Handler;
import android.os.Message;
import android.util.Log;

public class FlashTread implements Runnable {
    private Handler handler;
    private boolean flashRun = true;
    private int speed = 1000;
    FlashFragment flashFragment = new FlashFragment();

    public FlashTread(Handler handler){
        this.handler = handler;
    }
    @Override
    public void run()  {
        boolean toggleControl = false;
        while (flashRun) {
            try {
                Message message = handler.obtainMessage();
                if (toggleControl)
                    message.obj = "off";
                else
                    message.obj = "on";
                // handler.removeCallbacksAndMessages(null);
                handler.sendMessage(message);
                Thread.sleep(speed);
                toggleControl = !(toggleControl);
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
                System.out.println("Runnable interrupted");
            }

        }

    }
    public void setFlashRun(){
        this.flashRun = true;
    }
    public void setFlashRunStop(){ this.flashRun = false; }

    public void stop() {
        this.flashRun = false;
    }
    public void start() {
        this.flashRun = true;
    }
    public void setSpeed(int speed){
        this.speed = speed;
    }
}

File: app/src/main/java/com/cookandroid/torchappkt/FlashTread.kt
package com.cookandroid.torchappkt

import android.os.Handler

class FlashTread(private val handler: Handler) : Runnable {
    private var flashRun = true
    private var speed = 1000

    override fun run() {
        var toggleControl = false
        while (flashRun) {
            try {
                val message = handler.obtainMessage()
                if (toggleControl) message.obj = "off"
                else message.obj = "on"
                // handler.removeCallbacksAndMessages(null);
                handler.sendMessage(message)
                Thread.sleep(speed.toLong())
                toggleControl = !(toggleControl)
            } catch (e: InterruptedException) {
                Thread.currentThread().interrupt()
            }
        }
    }

    fun setFlashRun() {
        this.flashRun = true
    }

    fun setFlashRunStop() {
        this.flashRun = false
    }

    fun stop() {
        this.flashRun = false
    }

    fun start() {
        this.flashRun = true
    }

    fun setSpeed(speed: Int) {
        this.speed = speed
    }
}
